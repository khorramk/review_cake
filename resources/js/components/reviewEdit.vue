<template>
        <form class="edit-form form-container" @submit.prevent="editReview">
            <textarea class="edit-form__body" name="body" id="" cols="30" rows="10" v-model="reviewsEdits"></textarea>
            <input class="edit-form__submit-btn" :disabled="$store.state.isDisable" type="submit" value="edit review">
        </form>
</template>
<script>
export default {
    data() {
        return {
            reviewsEdits: ''
        }
    },
     //editing comments
    methods: {
       editReview() {
            this.$store.commit('disableButton');
            axios.put(`/api/reviews/${this.$route.params.id}`, {
                reviewId: this.$route.params.id,
                reviewsEdits: this.$data.reviewsEdits
            }).then(() => window.location.href = '/reviews');
    },
    },
}
</script>
