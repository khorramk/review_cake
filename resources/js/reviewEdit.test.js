

import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import reviewEdit from './components/reviewEdit';

const vm = new Vue(reviewEdit);

test('type of data for reviews edits should be in string', () => {
    expect(vm.reviewsEdits).toBe('')
});

test('disable the button after submit request', () => {
    vm.$on('submit', ()=> {
        expect(vm.isDisable).toBe(true)
    })
});
it('renders correct elements', () => {
    
    vm.$nextTick().then(()=> {
 
     expect(vm).toMatchSnapshot();
    }).catch().finally();
  })