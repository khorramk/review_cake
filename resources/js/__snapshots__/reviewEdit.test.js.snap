// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct elements 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "components": Object {},
    "data": [Function],
    "directives": Object {},
    "filters": Object {},
    "methods": Object {
      "editReview": [Function],
    },
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {
    "isDisable": false,
    "reviewsEdits": "",
  },
  "_directInactive": false,
  "_events": Object {
    "submit": Array [
      [Function],
    ],
  },
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "components": Object {},
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "methods": Object {
        "editReview": [Function],
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {
      "isDisable": false,
      "reviewsEdits": "",
    },
    "_directInactive": false,
    "_events": Object {
      "submit": Array [
        [Function],
      ],
    },
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "editReview": [Function],
    "isDisable": false,
    "reviewsEdits": "",
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 0,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "editReview": [Function],
  "isDisable": false,
  "reviewsEdits": "",
}
`;
